src/System.Management.Automation/security/SecuritySupport.cs

```diff
@@ -1336,35 +1337,15 @@ public enum ResolutionPurpose
 
     internal static class AmsiUtils
     {
-        private static string GetProcessHostName(string processName)
-        {
-            return string.Concat("PowerShell_", processName, ".exe_0.0.0.0");
-        }
-
         internal static int Init()
         {
             Diagnostics.Assert(s_amsiContext == IntPtr.Zero, "Init should be called just once");
 
             lock (s_amsiLockObject)
             {
-                Process currentProcess = Process.GetCurrentProcess();
-                string hostname;
-                try
-                {
-                    hostname = string.Concat("PowerShell_", Environment.ProcessPath, "_",
-                        currentProcess.MainModule.FileVersionInfo.ProductVersion);
-                }
-                catch (ComponentModel.Win32Exception)
-                {
-                    // This exception can be thrown during thread impersonation (Access Denied for process module access).
-                    hostname = GetProcessHostName(currentProcess.ProcessName);
-                }
-                catch (FileNotFoundException)
-                {
-                    // This exception can occur if the file is renamed or moved to some other folder
-                    // (This has occurred during Exchange set up).
-                    hostname = GetProcessHostName(currentProcess.ProcessName);
-                }
+                Assembly currentAssembly = typeof(AmsiUtils).Assembly;
+                string productVersion = currentAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
```

Comment:
We do the same in  `static PSVersionInfo()` and I guess we do it _before_. If I correct we can use productVersion from PSVersionInfo (perhaps we need new internal property).

Reply:
That's exactly what I tried to do initially, but when I added a ProductVersion internal property to PSVersionInfo, weird issues started to occur. More specifically, the value of that property was seemingly getting "injected" in completely unrelated places.
Was that related to some reflection magic being used somewhere? Just guessing, I didn't investigate further.

Reply:
PowerShell doesn't use reflection. Perhaps you need clean and full recompile.

Reply:
My fault, those weird issues were caused by a debug console output that I forgot to remove. 😅
I've pushed a new version of the PR that retrieves the product version from the newly added `PSVersionInfo.ProductVersion` static property.